### ISSUE TITLE
[BUG] ed25519 keys can not be loaded by the ARC module

### ISSUE THREAD

**Issue by Avamander** [rspamd/rspamd#4417]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Read about bug reporting in general: https://rspamd.com/doc/faq.html#how-to-report-bugs-found-in-rspamd
    * Enabled relevant debugging logs: https://rspamd.com/doc/faq.html#how-to-debug-some-module-in-rspamd 
    * Checked the FAQs about Core files in case of fatal crash: https://rspamd.com/doc/faq.html#how-to-figure-out-why-rspamd-process-crashed
    * Tried ASAN package and obtained the ASAN report (if possible): https://rspamd.com/doc/faq.html#asan-builds 
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+user%3Arspamd
    * Checked that there is not already an experimental package or master branch

**Describe the bug**
Loading ed25519 ARC keys from a file fails.

### Steps to Reproduce

1. Create an ed25519 key using the rspamd utility
2. Configure the ARC module to use it
3. Key fails to load (error indicates that it's interpreted as an RSA key)

**Expected behavior**
Ed25519 keys are loaded by the ARC module

### Versions

3.4

### Additional Information

Previously discussed on Telegram and confirmed to be a bug.

**Comment by SoLoR1** [comment]
I was trying to make this work today, it also breaks fallback to rsa selector/key file. As soon as you add ed25519 key file to arc it fails so sign emails.

**Comment by phonon112358** [comment]
Same problem still in Rspamd 3.13.2: 
- 
- OS: Debian 13 "Trixie"
- Rspamd version: 3.13.2 (official rspamd.org repo, not Debian standard repo)
- OpenSSL version: 3.5.1
- LuaJIT enabled: TRUE

DKIM signing works perfectly with both rsa2048 and ed25519 selectors (with both keys present).
**ARC signing only works if only the RSA selector is configured.** As soon as an Ed25519 selector is also present (even with a correct PEM file generated via `rspamadm ...`), ARC signing fails completely with:

    lua_rsa_privkey_load_file: cannot open private key from file: /var/lib/rspamd/dkim/example.com/ed25519-2025.key, error:1E08010C:DECODER routines::unsupported
    arc.lua:544: cannot load private key for signing

The ARC module does **not fallback** to the RSA selector if the Ed25519 selector cannot be loaded.
No ARC signature is set for the domain at all in this case.
This differs from the DKIM module, which does fall back to any available/valid selector.

This is 100% reproducible (multi-selector setups with Ed25519 present always block ARC).

Happy to provide further details, test builds or logs if needed!

**Comment by SoLoR1** [comment]
I understand this issue is not priority, because even providers like google or microsoft are not supporting ed25519 dkim keys. I had double dkim signed for like a week and i noticed in dmarc reports ed25519 was always giving failed, got annoyed and removed ed25519 signing all together.

**Comment by phonon112358** [comment]
yes... Google and Microsoft use ed25519 for signing but not for validation. However, there are several providers like e.g. gmx that use ed25519 validation. 
And Rspamd also validates ed25519 correctly, it only doesn't sign ARC with ed25519.
Just to clarify: GMX (one of the largest mailbox providers in Germany) does validate ed25519 DKIM keys (I have confirmed this with real mail traffic).  
I'm not able to test ARC with ed25519 at all, since the current Rspamd bug prevents ARC signing with ed25519 keys.  
So for anyone forwarding to GMX (which is very common in Europe), having multi-selector (RSA+ed25519) ARC signing in Rspamd is not just a theoretical issue but a real-world deliverability concern.

**Comment by vstakhov** [comment]
> I'm not able to test ARC with ed25519 at all, since the current Rspamd bug prevents ARC signing with ed25519 keys.

I have not checked the arc path indeed. Only DKIM one.

**Comment by phonon112358** [comment]
Strange, for me, dkim_signing with ed25519 works very fine (at least if the provider validates it, otherwise the rsa fallback key is taken).

However, not in ARC.

**Comment by vstakhov** [comment]
Ok, let me check if cursor can automatically fix it, as I'm sure the problem is very narrow: arc module just uses something wrong when loading PEM keys.

**Comment by phonon112358** [comment]
Hereby, I confirm that this issue is resolved by PR https://github.com/rspamd/rspamd/pull/5684
